test/reviewController.test.js

```diff
@@ -58,70 +59,219 @@ describe('ReviewController API Tests', () => {
-    expect(response.body.count).toBe(0);
+      // Assert that the "count" field reflects the actual count of reviews (which is 0 in this case)
+      expect(response.body.count).toBe(0);
+    });
   });
+  describe('test the getSingleReview endpoint', () => {
+    //get single review
+    it('should test the getSingleReview endpoint - Success Case', async () => {
+      const { _id: reviewId } = await Review.findOne({});
+
+      const response = await request(app).get(`/api/v1/reviews/${reviewId}`);
+      expect(response.status).toBe(StatusCodes.OK);
+      expect(response.body).toHaveProperty('review');
+      //expect(response.body.review._id.toString()).toBe(reviewId.toString());//check if the response.body.review._id value matches what we expect // or
+      expect(response.body.review).toMatchObject(
+        JSON.parse(JSON.stringify(review))
+      ); //checks if the object on the right is a subset (ie: has at least some of the same fields and values) as the object on the left.
+    });
 
-  //get single review
-  it('should test the getSingleReview endpoint - Success Case', async () => {
-    const { _id: reviewId } = await Review.findOne({});
-
-    const response = await request(app).get(`/api/v1/reviews/${reviewId}`);
-    expect(response.status).toBe(StatusCodes.OK);
-    expect(response.body).toHaveProperty('review');
-    //expect(response.body.review._id.toString()).toBe(reviewId.toString());//check if the response.body.review._id value matches what we expect // or
-    expect(response.body.review).toMatchObject(
-      JSON.parse(JSON.stringify(review))
-    ); //checks if the object on the right is a subset (ie: has at least some of the same fields and values) as the object on the left.
+    it('should test the getSingleReview endpoint - Error Case (Not Found)', async () => {
+      const response = await request(app).get('/api/v1/reviews/errorReviewId');
+      expect(response.status).toBe(StatusCodes.NOT_FOUND);
+    });
   });
+  describe('test the get All Reviews for the particular album endpoint', () => {
+    //get all reviews for this particular album
+    it('should get reviews for a specific album', async () => {
+      const response = await request(app).get(
+        `/api/v1/reviews/album/${album._id}`
+      );
 
-  it('should test the getSingleReview endpoint - Error Case (Not Found)', async () => {
-    const response = await request(app).get('/api/v1/reviews/errorReviewId');
-    expect(response.status).toBe(StatusCodes.NOT_FOUND);
+      expect(response.statusCode).toBe(200);
+      expect(response.body).toHaveProperty('allProductReviews');
+      expect(response.body.allProductReviews).toHaveLength(1);
+      expect(response.body).toHaveProperty('count');
+      expect(response.body.count).toBe(1);
+      // If no reviews are found, allProductReviews should be an empty array and count should be 0
+      if (response.body.count === 0) {
+        expect(response.body.allProductReviews).toEqual([]);
+      }
```

Comment:
Generally for a test it's a "[code smell](https://en.wikipedia.org/wiki/Code_smell)" for me if I see an `if` statement. For example, in this case, we know that the request we made on line 119 will always have exactly 1 review. So the contents of this if statement will _never_ be executed which means it's not really testing anything.

If we want to test it, we should write test code (probably as a separate test for good "separation of concerns") that creates an Album with no reviews, then does a `/api/v1/reviews/album/${secondAlbum._id}` request and verifies the shape of the response.